components/constellation/constellation.rs

```diff
@@ -1922,74 +1960,105 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         }
     }
 
-    fn traverse_frame_to_pipeline(&mut self, frame_id: FrameId, next_pipeline_id: PipelineId) {
+    // https://html.spec.whatwg.org/multipage/#traverse-the-history
+    fn traverse_to_entry(&mut self, entry: FrameState) {
+        // Step 1.
+        let frame_id = entry.frame_id;
+        let pipeline_id = match entry.pipeline_id {
+            Some(pipeline_id) => pipeline_id,
+            None => {
+                // If there is no pipeline, then the document for this
+                // entry has been discarded, so we navigate to the entry
+                // URL instead. When the document has activated, it will
+                // traverse to the entry, but with the new pipeline id.
+                debug!("Reloading document {} for frame {}.", entry.url, frame_id);
+                // TODO: referrer?
```

Comment:
That is an interesting point. Step 1 of https://html.spec.whatwg.org/multipage/browsers.html#history-traversal just refers to the URL of the entry, not anything else, though the note does imply we should at least store the HTTP method. We may also want to store persisted user state, e.g. the scroll position as per https://html.spec.whatwg.org/multipage/browsers.html#an-entry-with-persisted-user-state. (cc @paulrouget because this is relevant to the discussions around load data in the embedding API).

Not sure what we should do right now, just do a URL right now and file an issue?